### ISSUE TITLE
One or more errors occurred. (Schema mismatch for score column 'Score': expected vector of two or more items of type Single, got Vector<Single, 1> Arg_ParamName_Name)

### ISSUE THREAD

**Issue by rzechu** [dotnet/machinelearning#6573]
**System Information (please complete the following information):**
 - OS & Version: Windows 11
 - ML.NET Version: ML.NET v0.20.0
 - .NET Version:  NET 6.0

**Describe the bug**
One or more errors occurred. (Schema mismatch for score column 'Score': expected vector of two or more items of type Single, got Vector<Single, 1> Arg_ParamName_Name)
on AutoML Multiclasstraining

```
SELECT  
CAST(DecimalColumn1 as real) as [DecimalColumn1], 
CAST(IntColumn2 as real) as [IntColumn2], 
CAST(IntColumn3 as real) as [IntColumn3], 
CAST(IntColumn4 as real) as [IntColumn4], 
cast([LongStringColumn1] as varchar(2000)) as [LongStringColumn1],  /*nvarchar(max) */
[Name], /*nvarchar(128) */
[ShortStringColumn3] as [ShortStringColumn3], /*nvarchar(128) */
cast([LongStringColumn3] as varchar(2000)) as [LabelColumn]  /*nvarchar(max) */
FROM Documents 
```
![image](https://user-images.githubusercontent.com/45091660/219575314-5246a738-d1c2-4e42-9c2a-8f90225fdfbd.png)

**To Reproduce**
1. MulticlassClassification
2. Loading data from MSSQL Database
```
var loader = MLContext.Data.CreateDatabaseLoader(columns.ToArray());
var dbSource = new DatabaseSource(SqlClientFactory.Instance, connectionString, sqlQuery);
var iDataView = loader.Load(dbSource);
```
```
columns
    [0]: "Name=DecimalColumn1      Type=Single"
    [1]: "Name=IntColumn2               Type=Single"
    [2]: "Name=IntColumn3               Type=Single"
    [3]: "Name=IntColumn4               Type=Single"
    [4]: "Name=LongStringColumn1  Type=String"
    [5]: "Name=Name                        Type=String"
    [6]: "Name=ShortStringColumn3 Type=String"
    [7]: "Name=LabelColumn             Type=String"
```

```
iDataView
    [0]: "Index=0 Name=DecimalColumn1     Type=Single IsHidden=False"
    [1]: "Index=1 Name=IntColumn2               Type=Single IsHidden=False"
    [2]: "Index=2 Name=IntColumn3               Type=Single IsHidden=False"
    [3]: "Index=3 Name=IntColumn4               Type=Single IsHidden=False"
    [4]: "Index=4 Name=LongStringColumn1  Type=String IsHidden=False"
    [5]: "Index=5 Name=Name                         Type=String IsHidden=False"
    [6]: "Index=6 Name=ShortStringColumn3 Type=String IsHidden=False"
    [7]: "Index=7 Name=LabelColumn             Type=String IsHidden=False"
```
iDataView.Preview()
![image](https://user-images.githubusercontent.com/45091660/219575936-ee584f58-a8dd-4bcd-8b80-3d71d91994ca.png)

3. 
![image](https://user-images.githubusercontent.com/45091660/219574125-863951d5-a87a-44cc-8ac1-ca96db0354d9.png)
```
var experiment = MLContext.Auto().CreateMulticlassClassificationExperiment(multiClassExperimentSettings);
var experimentResult = experiment.Execute(trainData: iDataView, labelColumnName: runExperimentDto.LabelColumnName, progressHandler: progressHandler);
```

Error:
```
Schema mismatch for score column 'Score': expected vector of two or more items of type Single, got Vector<Single, 1> (Parameter 'schema')
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task`1.GetResultCore(Boolean waitCompletionNotification)
   at Microsoft.ML.AutoML.AutoMLExperiment.Run()
   at Microsoft.ML.AutoML.MulticlassClassificationExperiment.Execute(IDataView trainData, ColumnInformation columnInformation, IEstimator`1 preFeaturizer, IProgress`1 progressHandler)
   at AI.AutoML.Trainers.MulticlassClassification.RunExperiment(IDataView trainingDataView, RunExperimentDto runExperimentDto) in C:\.....\_myapp.cs:line 63
```

InnerException
```
Schema mismatch for score column 'Score': expected vector of two or more items of type Single, got Vector<Single, 1> (Parameter 'schema')

   at Microsoft.ML.Data.MulticlassClassificationEvaluator.CheckScoreAndLabelTypes(RoleMappedSchema schema)
   at Microsoft.ML.Data.EvaluatorBase`1.CheckColumnTypes(RoleMappedSchema schema)
   at Microsoft.ML.Data.EvaluatorBase`1.Microsoft.ML.Data.IEvaluator.Evaluate(RoleMappedData data)
   at Microsoft.ML.Data.MulticlassClassificationEvaluator.Evaluate(IDataView data, String label, String score, String predictedLabel)
   at Microsoft.ML.MulticlassClassificationCatalog.<>c__DisplayClass6_0.<CrossValidate>b__0(CrossValidationResult x)
   at System.Linq.Enumerable.SelectArrayIterator`2.ToArray()
   at Microsoft.ML.AutoML.MulticlassClassificationRunner.Run(TrialSettings settings)
   at System.Threading.Tasks.Task`1.InnerInvoke()
   at System.Threading.Tasks.Task.<>c.<.cctor>b__272_0(Object obj)
   at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location ---
   at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
--- End of stack trace from previous location ---
   at Microsoft.ML.AutoML.AutoMLExperiment.<RunAsync>d__26.MoveNext()
```

![image](https://user-images.githubusercontent.com/45091660/219575172-ffb91264-bc70-4493-909b-99d294078210.png)

**Expected behavior**
This case should work (I am using same scenario for many other columns and only few cases gives this error - propably related to nvarchar(max) columns or some kind of mix columns
I am many times using  2 varchars + 2 ints as source with  int/varchar output and work without any doubts
Or should be understandable workaround

**Comment by LittleLittleCloud** [comment]
How many distinct values do you have for `LabelName` column? Based on exception message it seems that your label column only have one distinct value?

**Comment by rzechu** [comment]
> How many distinct values do you have for `LabelName` column? Based on exception message it seems that your label column only have one distinct value?

https://user-images.githubusercontent.com/45091660/219575314-5246a738-d1c2-4e42-9c2a-8f90225fdfbd.png

**Comment by LittleLittleCloud** [comment]
@rzechu 
Is it possible for you to export part of your SQL database as csv and give it to us? We want to try reproducing the issue you have on our side

**Comment by rzechu** [comment]
> @rzechu Is it possible for you to export part of your SQL database as csv and give it to us? We want to try reproducing the issue you have on our side
> 
```
CREATE TABLE [dbo].[AI](
	[DecimalColumn1] [decimal](20, 4) NULL,
	[IntColumn2] [int] NULL,
	[IntColumn3] [int] NULL,
	[IntColumn4] [int] NULL,
	[LongStringColumn1] [nvarchar](max) NULL,
	[Name] [nvarchar](255) NULL,
	[ShortStringColumn3] [nvarchar](128) NULL,
	[LabelColumn] [nvarchar](max) NULL
) 
GO
INSERT [dbo].[Ai] ([DecimalColumn1], [IntColumn2], [IntColumn3], [IntColumn4], [LongStringColumn1], [Name], [ShortStringColumn3], [LabelColumn]) VALUES (NULL, 73, 0, NULL, N'Paryż', N'Delegacja zagraniczna | Administrator Systemu', NULL, N'Konferencja')
GO
INSERT [dbo].[Ai] ([DecimalColumn1], [IntColumn2], [IntColumn3], [IntColumn4], [LongStringColumn1], [Name], [ShortStringColumn3], [LabelColumn]) VALUES (NULL, 8, 0, 1583, N'Warszawa', N'Delegacja krajowa | Alicja Konieczny', NULL, N'Szkolenie')
GO
INSERT [dbo].[Ai] ([DecimalColumn1], [IntColumn2], [IntColumn3], [IntColumn4], [LongStringColumn1], [Name], [ShortStringColumn3], [LabelColumn]) VALUES (NULL, 73, 0, 1583, N'Warszawa', N'Delegacja krajowa | Administrator System', NULL, N'Konferencja')
GO

SELECT 
CAST(DecimalColumn1 as real) as [DecimalColumn1], 
CAST(IntColumn2 as real) as [IntColumn2], 
CAST(IntColumn3 as real) as [IntColumn3], 
CAST(IntColumn4 as real) as [IntColumn4], 
[LongStringColumn1], 
[Name] as [Name], 
[ShortStringColumn3] as [ShortStringColumn3], 
[LabelColumn] as [LabelColumn] 
FROM IAi

```

**Comment by LittleLittleCloud** [comment]
If there's only three rows of dataset, it's possible that after divided the training set only contains one label value and validation set contains another.

To validate that assumption, can you try setting validation dataset using the same IDataView as training dataset which avoids the train-validation split by AutoML API.

**Comment by rzechu** [comment]
> If there's only three rows of dataset, it's possible that after divided the training set only contains one label value and validation set contains another. (training set contains row 0 and row 2, validation set contains row 1)
> 
> To validate that assumption, can you try setting validation dataset using the same IDataView as training dataset which avoids the train-validation split by AutoML API.

You are right! I used this SQL query


```
SELECT 
CAST(DecimalColumn1 as real) as [DecimalColumn1], 
CAST(IntColumn2 as real) as [IntColumn2], 
CAST(IntColumn3 as real) as [IntColumn3], 
CAST(IntColumn4 as real) as [IntColumn4], 
[LongStringColumn1], 
[Name] as [Name], 
[ShortStringColumn3] as [ShortStringColumn3], 
[LabelColumn] as [LabelColumn] 
FROM IAi

UNION ALL
SELECT
CAST(NULL as real) as [DecimalColumn1], 
CAST(8 as real) as [IntColumn2], 
CAST(0 as real) as [IntColumn3], 
CAST(NULL as real) as [IntColumn4], 
'Kraków' as [Doc_LongStringColumn1], 
'Delegacja krajowa | Alicja Konieczny' as [Name], 
NULL as [ShortStringColumn3], 
'Szkolenie' as [LabelColumn]

UNION ALL
SELECT
CAST(NULL as real) as [DecimalColumn1], 
CAST(8 as real) as [IntColumn2], 
CAST(0 as real) as [IntColumn3], 
CAST(NULL as real) as [IntColumn4], 
'Kraków' as [Doc_LongStringColumn1], 
'Delegacja krajowa | Alicja Konieczny' as [Name], 
NULL as [ShortStringColumn3], 
'Szkolenie' as [LabelColumn]

UNION ALL
SELECT
CAST(NULL as real) as [DecimalColumn1], 
CAST(8 as real) as [IntColumn2], 
CAST(0 as real) as [IntColumn3], 
CAST(NULL as real) as [IntColumn4], 
'Kraków' as [Doc_LongStringColumn1], 
'Delegacja krajowa | Alicja Konieczny' as [Name], 
NULL as [ShortStringColumn3], 
'Szkolenie' as [LabelColumn]

UNION ALL
SELECT
CAST(NULL as real) as [DecimalColumn1], 
CAST(73 as real) as [IntColumn2], 
CAST(0 as real) as [IntColumn3], 
CAST(NULL as real) as [IntColumn4], 
'Berlin' as [Doc_LongStringColumn1], 
'Delegacja zagraniczna | Administrator Systemu' as [Name], 
NULL as [ShortStringColumn3], 
'Konferencja' as [LabelColumn]
```
![image](https://user-images.githubusercontent.com/45091660/223045738-a77a4ef7-8e74-4e55-ac34-ada4b9eae0d0.png)

And training was done without any doubts

**Comment by rosebyte** [comment]
Closed as answered.
